### PR TITLE
fix: star routing fixes

### PR DESCRIPTION
**Pull request by alaeddine-13**
A few fixes after reviewing the tutorials and docs
closes: https://github.com/jina-ai/internal-tasks/issues/342

### CODE REVIEW

File: jina/types/request/__init__.py [hunk 1]
```diff
@@ -24,6 +24,8 @@ class Request(ProtoTypeMixin):
     """
 
     def __getattr__(self, name: str):
+        if name == 'non_empty_fields':
```
Comment (JoanFM) [inline jina/types/request/__init__.py:27]: can u add a test for this?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/types/request/__init__.py (hunk 1 above)
Comment (alaeddine-13) [reply jina/types/request/__init__.py:27]: added

Comment (alaeddine-13) [reply jina/types/request/__init__.py]: non_empty_fields was completely removed from Request in https://github.com/jina-ai/jina/pull/4102

**Review by jacobowitz — APPROVED** [review]
nice docs improvements :+1:

**Pull request merged**